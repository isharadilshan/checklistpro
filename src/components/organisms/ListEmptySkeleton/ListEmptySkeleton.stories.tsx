import React from 'react';
import {NativeBaseProvider} from 'native-base';
import {Provider} from 'react-redux';
import {ComponentMeta, ComponentStory} from '@storybook/react';
import redux from '../../../redux/store';

import ListEmptySkeleton from './';

export default {
  title: 'components/organisms/ListEmptySkeleton',
  component: ListEmptySkeleton,
} as ComponentMeta<typeof ListEmptySkeleton>;

export const Basic: ComponentStory<typeof ListEmptySkeleton> = () => {
  return (
    <Provider store={redux.store}>
      <NativeBaseProvider>
        <ListEmptySkeleton />
      </NativeBaseProvider>
    </Provider>
  );
};
