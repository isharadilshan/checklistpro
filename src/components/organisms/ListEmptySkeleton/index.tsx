import React from 'react';
import {HStack, Skeleton, VStack} from 'native-base';
import createStyle from './styles';

const SkeletonTemplate = () => {
  const styles = createStyle();

  return (
    <HStack maxW="400" space={8} p={4} my={2} style={styles.skeletonWrapper}>
      <Skeleton flex="1" h="150" rounded="md" startColor="coolGray.700" />
      <VStack flex="3" space="4">
        <Skeleton startColor="coolGray.700" />
        <Skeleton.Text startColor="coolGray.700" />
        <HStack space="2" alignItems="center">
          <Skeleton h="3" startColor="coolGray.700" rounded="full" />
        </HStack>
      </VStack>
    </HStack>
  );
};

const ListEmptySkeleton: React.FC = () => {
  return (
    <>
      <SkeletonTemplate />
      <SkeletonTemplate />
      <SkeletonTemplate />
      <SkeletonTemplate />
      <SkeletonTemplate />
    </>
  );
};

export default ListEmptySkeleton;
