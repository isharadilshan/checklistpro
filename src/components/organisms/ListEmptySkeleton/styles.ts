import {ViewStyle, StyleSheet} from 'react-native';

interface Style {
  skeletonWrapper: ViewStyle;
}

export default () => {
  return StyleSheet.create<Style>({
    skeletonWrapper: {
      width: '100%',
      borderWidth: 1,
      borderRadius: 4,
      borderColor: '#6b7280',
    },
  });
};
