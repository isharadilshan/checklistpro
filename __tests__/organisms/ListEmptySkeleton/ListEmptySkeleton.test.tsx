import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import ListEmptySkeleton from '../../../src/components/organisms/ListEmptySkeleton';

configure({adapter: new Adapter()});

describe('when rendering', () => {
  const wrapper = shallow(<ListEmptySkeleton />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
