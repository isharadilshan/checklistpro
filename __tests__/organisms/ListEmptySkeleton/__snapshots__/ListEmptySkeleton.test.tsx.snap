// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<Fragment>
  <SkeletonTemplate />
  <SkeletonTemplate />
  <SkeletonTemplate />
  <SkeletonTemplate />
  <SkeletonTemplate />
</Fragment>
`;
